backend/api/system/authentication.py

```diff
@@ -123,6 +149,71 @@ def get(self):
             return {"Error": "Unable to authenticate", "SubCode": "AuthError"}, 500
 
 
+class OSMTeamsAuthenticationCallbackAPI(Resource):
+    def get(self):
+        """
+        Handles the OSM Teams OAuth callback
+        ---
+        tags:
+          - system
+        produces:
+          - application/json
+        parameters:
+            - in: query
+              name: redirect_uri
+              description: Route to redirect user once authenticated
+              type: string
+              default: /take/me/here
+              required: false
+            - in: query
+              name: code
+              description: Code obtained after user authorization
+              type: string
+              required: true
+            - in: query
+              name: email_address
+              description: Email address to used for email notifications from TM.
+              type: string
+              required: false
+        responses:
+          302:
+            description: Redirects to login page, or login failed page
+          500:
+            description: A problem occurred authenticating the user
+          502:
+            description: A problem occurred negotiating with the OSM API
+        """
+
+        token_url = f"{EnvironmentConfig.OSM_TEAMS_AUTH_URL}/hyauth/oauth2/token"
```

Comment:
@willemarcel in some environments where there are internal DNS and external urls, the TOKEN URL and the AUTH URL are different. I think we should set it like this:

`OSM_TEAMS_AUTH_URL` with default `auth.mapping.team/hyauth/oauth2/auth` and OSM_TEAMS_TOKEN_URL with default `auth.mapping.team/hyauth/oauth2/token`